test/integration/health_test.go

```diff
@@ -44,3 +48,18 @@ func TestHealthCheck(t *testing.T) {
 	assert.NoError(t, err)
 	assert.Equal(t, resp.Status, healthpb.HealthCheckResponse_SERVING)
 }
+
+func TestHTTPHealthCheck(t *testing.T) {
+	// use HTTP health check
+	resp, err := http.Get("http://" + defaultServer.RPCAddr() + "/healthz/")
+	defer func() {
+		assert.NoError(t, resp.Body.Close())
+	}()
+	assert.NoError(t, err)
+	assert.Equal(t, resp.StatusCode, 200)
+
+	var healthResp health.CheckResponse
+	err = json.NewDecoder(resp.Body).Decode(&healthResp)
+	assert.NoError(t, err)
+	assert.Equal(t, healthResp.Status, grpchealth.StatusServing.String())
```

Comment:
I was thinking of doing this, but is it bad?
```go
	for _, serviceName := range serviceNames {
		t.Run(serviceName, func(t *testing.T) {
			resp, err := cli.Check(context.Background(), &healthpb.HealthCheckRequest{
				Service: serviceName,
			})
			assert.NoError(t, err)
			assert.Equal(t, resp.Status, healthpb.HealthCheckResponse_SERVING)
		})
	}
```

Reply:
This approach is also good.
But I think showcasing the actual use cases by explicitly showing service name will be better. Like `PushPull` ...

Reply:
If I'm going to create a health check test for each service, wouldn't it be nice to cover the unknown service?
```go
func TestRPCHealthCheckUnknownService(t *testing.T) {
	// use gRPC health check for unknown service
	conn, err := grpc.Dial(
		defaultServer.RPCAddr(),
		grpc.WithTransportCredentials(insecure.NewCredentials()),
	)
	assert.NoError(t, err)
	defer func() {
		assert.NoError(t, conn.Close())
	}()

	cli := healthpb.NewHealthClient(conn)
	_, err = cli.Check(context.Background(), &healthpb.HealthCheckRequest{
		Service: "unknown",
	})
	assert.Error(t, err)
}
```

And can you suggest a name for the test function?
I'm thinking TestRPCHealthCheckYorkie, TestRPCHealthCheckAdmin.

Reply:
> If I'm going to create a health check test for each service, wouldn't it be nice to cover the unknown service?

Yes, it will be good to also consider unknown service.

> And can you suggest a name for the test function?
I'm thinking TestRPCHealthCheckYorkie, TestRPCHealthCheckAdmin.

Looks good to me.

Reply:
I don't know if I did what you wanted 🥲.
I separated the tests by service, but merged the default health check and the unknown health check.
How about wrapping the repeated parts of each test into a function?